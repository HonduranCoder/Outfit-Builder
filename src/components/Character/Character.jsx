import React from 'react';

export default function Character({ face, top, bottom }) {
  return (
    <section style={{ display: 'flex', flexDirection: 'column' }}>
      <article
        className="face"
        aria-label="face"
        style={{
          backgroundImage: `url(./${face}-head.png)`,
          height: 150,
          width: 150,
          backgroundSize: 'cover',
          backgroundRepeat: 'no-repeat',
        }}
      ></article>

      <article
        className="top"
        aria-label="top"
        style={{
          backgroundImage: `url(./${top}-top.png)`,
          height: 150,
          width: 150,
          backgroundSize: 'cover',
          backgroundRepeat: 'no-repeat',
        }}
      ></article>

      <section style={{ display: 'flex', flexDirection: 'column' }}>
        <article
          className="bottom"
          aria-label="bottom"
          style={{
            backgroundImage: `url(./${bottom}-bottom.png)`,
            height: 150,
            width: 150,
            backgroundSize: 'cover',
            backgroundRepeat: 'no-repeat',
          }}
        ></article>
      </section>
    </section>
  );
}
