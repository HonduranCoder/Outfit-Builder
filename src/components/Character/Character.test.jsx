import { screen, render } from '@testing-library/react';
import Character from './Character';

it('should render pieces of character', () => {
  const { container } = render(
    <Character face="smiley" top="tank" bottom="shorts" />
  );
  const face = screen.getByLabelText('face');
  const top = screen.getByLabelText('top');
  const bottom = screen.getByLabelText('bottom');

  expect(face).toHaveStyle(`background-image: url(./smiley-head.png)`);
  expect(top).toHaveStyle(`background-image: url(./tank-top.png)`);
  expect(bottom).toHaveStyle(`background-image: url(./shorts-bottom.png)`);

  expect(container).toMatchSnapshot();
});
