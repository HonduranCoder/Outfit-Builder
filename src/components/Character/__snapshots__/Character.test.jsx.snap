// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render pieces of character 1`] = `
<div>
  <section
    style="display: flex; flex-direction: column;"
  >
    <article
      aria-label="face"
      class="face"
      style="background-image: url(./smiley-head.png); height: 150px; width: 150px; background-size: cover; background-repeat: no-repeat;"
    />
    <article
      aria-label="top"
      class="top"
      style="background-image: url(./tank-top.png); height: 150px; width: 150px; background-size: cover; background-repeat: no-repeat;"
    />
    <section
      style="display: flex; flex-direction: column;"
    >
      <article
        aria-label="bottom"
        class="bottom"
        style="background-image: url(./shorts-bottom.png); height: 150px; width: 150px; background-size: cover; background-repeat: no-repeat;"
      />
    </section>
  </section>
</div>
`;
